JLio.Commands/Advanced/Compare.cs

```diff
@@ -0,0 +1,482 @@
+        {
+            var result = new CompareResults();
+            if (source.Type != target.Type)
+            {
+                result.Add(GetDifferentTypeResult(source, target));
+            }
+            else
+            {
+                result.Add(GetSameTypeResult(source, target));
+                result.AddRange(CompareSameTypeObjects(source, target));
+            }
+
+            return result;
+        }
+
+        private CompareResults CompareSameTypeObjects(JToken source, JToken target)
+        {
+            var result = new CompareResults();
+            switch (source.Type)
+            {
+                case JTokenType.Boolean:
+                    result.AddRange(CompareBoolean(source, target));
+                    break;
+                case JTokenType.Array:
+                    result.AddRange(CompareArray((JArray) source, (JArray) target));
+                    break;
+                case JTokenType.Float:
+                    result.AddRange(CompareIntegerNumbers(source, target));
+                    break;
+                case JTokenType.Integer:
+                    result.AddRange(CompareFloatNumbers(source, target));
+                    break;
+                case JTokenType.Object:
+                    result.AddRange(CompareObjectProperties((JObject) source, (JObject) target));
+                    break;
+                default:
+                    result.AddRange(CompareDeepEquals(source, target));
+                    break;
+            }
+
+            return result;
+        }
+
+        private IEnumerable<CompareResult> CompareObjectProperties(JObject source, JObject target)
+        {
+            var result = new CompareResults();
+            var allProperties = new List<string>();
+            allProperties.AddRange(GetPropertyNames(source));
+            allProperties.AddRange(GetPropertyNames(target));
+
+            foreach (var property in allProperties.Distinct())
+                result.AddRange(CompareProperty(source, target, property));
+            return result;
+        }
+
+        private CompareResults CompareProperty(JObject source, JObject target, string propertyName)
+        {
+            var result = new CompareResults();
+            var sourceToken = source[propertyName];
+            var targetToken = target[propertyName];
+            if (sourceToken == null || targetToken == null)
+                result.Add(GetStructureDifferenceResult(source, target, propertyName));
+            else
+                result.AddRange(CompareTokens(sourceToken, targetToken));
+            return result;
+        }
+
+        private CompareResult GetStructureDifferenceResult(JObject source, JObject target, string propertyName)
+        {
+            var sourcePath = GetPath(source, options.ItemsFetcher);
+            var targetPath = GetPath(target, options.ItemsFetcher);
+            return new CompareResult
```

Comment:
consider having static factory methods on the CompareResult class to help reduce some of the noise from this command.
If you have something like:  return CompareResult.StructureDifference(source, target)   
this could help improve readibility, and also not have to provide all of these very specific values like DifferenceType, DifferenceSubType, Description, FoundDifference - because all of these will be known by the factory method.

Do the same for the other types of differences too.